### PR TITLE
Complete merge-ort implementation...almost

### PR DESCRIPTION
**Pull request by newren**
This series (nearly) completes the merge-ort implementation, cleaning up testsuite failures.  The exceptions are some t6423 failures being addressed independently[1].

Changes since v1, to address feedback from Ævar:
  * Squashed patches 3 & 4
  * Various style fixes
  * Check attr_index->initialized before calling initialize_attr_index() instead of inside initialize_attr_index()
  * Simplify GIT_TEST_MERGE_ALGORITHM comparison to only have if-then block

Stuff not included in v2:
  * Ævar suggested patching the test expectation on the 4 known failing tests so that all tests passed under GIT_TEST_MERGE_ALGORITHM=ort.  Seems reasonable, but the semantic conflicts with other series might make it more trouble than it's worth and other series will fix all 4 tests.  Leaving as-is for now to avoid putting more burden on Junio; see https://lore.kernel.org/git/CABPp-BHeR6m4-M=nSX5NZtA2js3E3EVbAyDSMtp3-rN2QnUjqw@mail.gmail.com/
  * Ævar noted a few bigger cleanups to surrounding code that could also be done, even if orthogonal to this series.  I'll leave those for other series to address.

[1] See https://lore.kernel.org/git/pull.845.git.1614484707.gitgitgadget@gmail.com/ and batch 10 and 12 from https://github.com/gitgitgadget/git/pulls?q=is%3Apr+author%3Anewren+Optimization+batch

CC: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
CC: Jonathan Nieder <jrnieder@gmail.com>
CC: Derrick Stolee <dstolee@microsoft.com>
CC: Junio C Hamano <gitster@pobox.com>
cc: Elijah Newren <newren@gmail.com>
cc: Derrick Stolee <stolee@gmail.com>

### CODE REVIEW

**Review by priestesson — APPROVED** [review]

**Review by priestesson — APPROVED** [review]

**Comment by shamerdarock** [comment]
Filed changed

**Comment by turki0l0** [comment]
Thanks

**Comment by turki0l0** [comment]
Thanks

**Review by turki0l0 — APPROVED** [review]

File: t/t6428-merge-conflicts-sparse.sh [hunk 1]
```diff
@@ -0,0 +1,158 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (turki0l0) [reply t/t6428-merge-conflicts-sparse.sh:1]: t/t6428-merge-conflicts-sparse.sh

**Review by turki0l0 — APPROVED** [review]
Thanks

**Review by turki0l0 — APPROVED** [review]
Stuff not included in v2:

**Comment by newren** [comment]
Replaced by the ort-readiness pull request.

**Pull request closed without merging**